src/main/java/com/fauna/query/QueryOptions.java

```diff
@@ -5,66 +5,139 @@
+     *
+     * @param builder the builder with values for query options.
+     */
+    public QueryOptions(final Builder builder) {
         this.linearized = builder.linearized;
         this.typeCheck = builder.typeCheck;
         this.timeout = builder.timeout;
         this.queryTags = builder.queryTags;
         this.traceParent = builder.traceParent;
     }
 
+    /**
+     * Default QueryOptions instance with default configurations.
+     *
+     * @return a new QueryOptions instance with defaults.
+     */
+    public static QueryOptions getDefault() {
+        return QueryOptions.builder().build();
+    }
+
+    /**
+     * Returns an Optional indicating if linearized reads are enabled.
+     *
+     * @return an Optional containing the linearized setting, or empty if not
+     * specified.
+     */
     public Optional<Boolean> getLinearized() {
         return Optional.ofNullable(this.linearized);
     }
 
+    /**
+     * Returns an Optional indicating if type checking is enabled.
+     *
+     * @return an Optional containing the typeCheck setting, or empty if not
+     * specified.
+     */
     public Optional<Boolean> getTypeCheck() {
         return Optional.ofNullable(this.typeCheck);
     }
 
+    /**
+     * Returns an Optional of the timeout duration in milliseconds.
+     *
+     * @return an Optional containing the timeout duration in milliseconds, or
+     * empty if not specified.
+     */
     public Optional<Long> getTimeoutMillis() {
         return Optional.ofNullable(this.timeout).map(Duration::toMillis);
     }
 
+    /**
+     * Returns an Optional of the query tags.
+     *
+     * @return an Optional containing the QueryTags, or empty if not specified.
+     */
     public Optional<QueryTags> getQueryTags() {
         return Optional.ofNullable(this.queryTags);
     }
 
+    /**
+     * Returns an Optional of the trace parent for distributed tracing.
+     *
+     * @return an Optional containing the traceParent, or empty if not
+     * specified.
+     */
     public Optional<String> getTraceParent() {
         return Optional.ofNullable(this.traceParent);
     }
 
+    /**
+     * Builder class for constructing instances of QueryOptions.
+     */
     public static class Builder {
-        public Boolean linearized = null;
-        public Boolean typeCheck = null;
-        public Duration timeout = DEFAULT_TIMEOUT;
-        public QueryTags queryTags = null;
-        public String traceParent = null;
-
-        public Builder linearized(boolean linearized) {
+        private Boolean linearized = null;
+        private Boolean typeCheck = null;
+        private Duration timeout = DEFAULT_TIMEOUT;
+        private QueryTags queryTags = null;
+        private String traceParent = null;
+
+        /**
+         * Sets whether reads are linearized.
+         *
+         * @param linearized true to enable linearized reads, false otherwise.
+         * @return this Builder instance for chaining.
+         */
```

Comment:
```suggestion
        /**
         * If true, read-only transactions that don't read indexes are <a
         * href="https://docs.fauna.com/fauna/current/learn/transactions/">strictly
         * serialized</a>.
         * @param linearized true to enable linearized reads, false otherwise.
         * @return this Builder instance for chaining.
         */
```